great_expectations/execution_engine/pandas_execution_engine.py

```diff
@@ -353,6 +355,10 @@ def get_batch_data_and_markers(  # noqa: C901, PLR0912, PLR0915
             else:
                 df = reader_fn_result
 
+        elif isinstance(batch_spec, FabricBatchSpec):
+            reader_fn = batch_spec.get_reader_function()
+            df = reader_fn(**batch_spec.reader_options)
+
         else:
             raise gx_exceptions.BatchSpecError(
                 f"""batch_spec must be of type RuntimeDataBatchSpec, PandasBatchSpec, PathBatchSpec, S3BatchSpec, or AzureBatchSpec, \
```

Comment:
Should we update this exception with the new type?

Reply:
Done